index.js

```diff
@@ -76,24 +78,32 @@ class Client {
             if (exists) {
               this.serverless.cli.log(`Deleting all objects from bucket...`);
               return bucketUtils
-                .emptyBucket(this.aws, bucketName)
+                .emptyBucket(this.aws, bucketName, keyPrefix)
                 .then(() => {
-                  if (manageResources === false) {
-                    this.serverless.cli.log(
-                      'manageResources has been set to "false". Bucket will not be deleted'
-                    );
+                  if (keyPrefix) {
+                    this.serverless.cli.log(`Removed only the files under the prefix ${keyPrefix}`);
+                    return true;
                   } else {
-                    this.serverless.cli.log(`Removing bucket...`);
-                    return bucketUtils.deleteBucket(this.aws, bucketName);
+                    if (manageResources === false) {
+                      this.serverless.cli.log(
+                        'manageResources has been set to "false". Bucket will not be deleted'
+                      );
+                    } else {
+                      this.serverless.cli.log(`Removing bucket...`);
+                      return bucketUtils.deleteBucket(this.aws, bucketName);
+                    }
                   }
                 })
                 .then(() => {
+
                   if (manageResources === false) {
                     this.serverless.cli.log(`Success! Your files have been removed`);
                   } else {
-                    this.serverless.cli.log(
-                      `Success! Your files have been removed and your bucket has been deleted`
-                    );
+                    if (!keyPrefix) {
+                      this.serverless.cli.log(
+                        `Success! Your files have been removed and your bucket has been deleted`
```

Comment:
Can you add a logging statement here for if(keyPrefix) 

Let's give the user clarification that files from the keyPrefix are deleted.